### PR TITLE
feat: add front end support for WPA Enterprise

### PR DESCRIPTION
**Pull request by GregoryIvo**
> **Note**: We are using the Conventional Commits convention for our pull request titles. Please take a look at the [PR title format document](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#submitting-the-changes) for the supported [types](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#type) and [scopes](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#scope).

Brief description of the PR. [e.g. Added `null` check on `object` to avoid `NullPointerException`]

**Related Issue:** This PR fixes/closes {issue number}

**Description of the solution adopted:** A more detailed description of the changes made to solve/close one or more issues. If the PR is simple and easy to understand this section can be skipped

**Screenshots:** If applicable, add screenshots to help explain your solution

**Manual Tests**: Optional description of the tests performed to check correct functioning of changes, useful for an efficient review

**Any side note on the changes made:** Description of any other change that has been made, which is not directly linked to the issue resolution [e.g. Code clean up/Sonar issue resolution]

### CODE REVIEW

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/model/Gwt8021xConfig.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
+/*******************************************************************************
+ * Copyright (c) 2011, 2023 Eurotech and/or its affiliates and others
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/model/Gwt8021xConfig.java]: I think it is just 2023

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/model/Gwt8021xEap.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+ * Copyright (c) 2011, 2020 Eurotech and/or its affiliates and others
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/model/Gwt8021xEap.java]: Fix dates

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net/GwtNetworkServiceImpl.java [hunk 3]
```diff
@@ -353,16 +353,23 @@ private static List<GwtNetInterfaceConfig> privateFindNetInterfaceConfigurations
...
+                                        gwtWifiConfig.setSecurity(GwtWifiSecurity.netWifiSecurityWPA2.name());
+                                        break;
+                                    case SECURITY_WPA_WPA2:
+                                        gwtWifiConfig.setSecurity(GwtWifiSecurity.netWifiSecurityWPA_WPA2.name());
+                                        break;
+                                    case SECURITY_WEP:
+                                        gwtWifiConfig.setSecurity(GwtWifiSecurity.netWifiSecurityWEP.name());
+                                        break;
+                                    case SECURITY_NONE:
+                                    default:
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net/GwtNetworkServiceImpl.java:370]: If, in the future, we add a new type to the enum then it is not correct to default to `NONE`. Either:
- add an enum value to `GwtWifiSecurity` for "unsupported" and handle it; or
- restrict the imported `kura.net.wifi` to `[2.4,2.5)`.

I'd prefer the second one.

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/Tab8021xUi.java [hunk 4]
```diff
@@ -0,0 +1,355 @@
+/*******************************************************************************
+ * Copyright (c) 2011, 2023 Eurotech and/or its affiliates and others
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/Tab8021xUi.java]: Fix dates

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net/GwtNetworkServiceImpl.java (hunk 3 above)
Comment (GregoryIvo) [reply kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net/GwtNetworkServiceImpl.java:370]: I went with option 2

**Review by marcellorinaldo — COMMENTED** [review]
With this configuration:

- Enterprise EAP = any value TLS/TTLS/PEAP
- Inner Authentication = NONE
- Identity = test
- Password = hellotesttest

I get the following stacktrace:

```java
2023-09-14T09:58:09,736 [ConfigurationListener Event Queue] ERROR o.e.k.n.NMDbusConnector - Unable to apply configuration to the device path /org/freedesktop/NetworkManager/Devices/3
java.util.NoSuchElementException: The "net.interface.wlan0.config.802-1x.client-cert" key is missing.
	at org.eclipse.kura.nm.NetworkProperties.get(NetworkProperties.java:41)
	at org.eclipse.kura.nm.configuration.NMSettingsConverter.create8021xTls(NMSettingsConverter.java:169)
	at org.eclipse.kura.nm.configuration.NMSettingsConverter.build8021xSettings(NMSettingsConverter.java:126)
	at org.eclipse.kura.nm.configuration.NMSettingsConverter.buildSettings(NMSettingsConverter.java:97)
	at org.eclipse.kura.nm.NMDbusConnector.enableInterface(NMDbusConnector.java:443)
	at org.eclipse.kura.nm.NMDbusConnector.manageConfiguredInterface(NMDbusConnector.java:422)
	at org.eclipse.kura.nm.NMDbusConnector.doApply(NMDbusConnector.java:381)
	at org.eclipse.kura.nm.NMDbusConnector.lambda$0(NMDbusConnector.java:366)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at org.eclipse.kura.nm.NMDbusConnector.doApply(NMDbusConnector.java:363)
	at org.eclipse.kura.nm.NMDbusConnector.apply(NMDbusConnector.java:322)
	at org.eclipse.kura.nm.configuration.NMConfigurationServiceImpl.writeNetworkConfigurationSettings(NMConfigurationServiceImpl.java:317)
	at org.eclipse.kura.nm.configuration.NMConfigurationServiceImpl.update(NMConfigurationServiceImpl.java:218)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.felix.scr.impl.inject.methods.BaseMethod.invokeMethod(BaseMethod.java:228)
	at org.apache.felix.scr.impl.inject.methods.BaseMethod.access$500(BaseMethod.java:41)
	at org.apache.felix.scr.impl.inject.methods.BaseMethod$Resolved.invoke(BaseMethod.java:664)
	at org.apache.felix.scr.impl.inject.methods.BaseMethod.invoke(BaseMethod.java:510)
	at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:317)
	at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:307)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.invokeModifiedMethod(SingleComponentManager.java:836)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.modify(SingleComponentManager.java:791)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.reconfigure(SingleComponentManager.java:709)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.reconfigure(SingleComponentManager.java:673)
	at org.apache.felix.scr.impl.manager.ConfigurableComponentHolder.configurationUpdated(ConfigurableComponentHolder.java:435)
	at org.apache.felix.scr.impl.manager.RegionConfigurationSupport.configurationEvent(RegionConfigurationSupport.java:316)
	at org.apache.felix.scr.impl.manager.RegionConfigurationSupport$2.configurationEvent(RegionConfigurationSupport.java:118)
	at org.eclipse.equinox.internal.cm.EventDispatcher$1.run(EventDispatcher.java:92)
	at org.eclipse.equinox.internal.cm.SerializedTaskQueue$1.run(SerializedTaskQueue.java:40)
```

Maybe such combination is not allowed? If not allowed, we should prevent the user from setting it by disabling some fields.

Setting Inner Authentication to `MSCHAPv2` does not cause issues.

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/Tab8021xUi.java [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+
+    @Override
+    public boolean isDirty() {
+        return this.dirty;
+    }
+
+    @Override
+    public boolean isValid() {
+        return true;
+    }
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/Tab8021xUi.java]: We are allowing to set an empty password when filling the username. This leads to the following error in the `kura.log`:

```
java.util.NoSuchElementException: The "net.interface.wlan0.config.802-1x.password" key contains an empty string value.
	at org.eclipse.kura.nm.NetworkProperties.get(NetworkProperties.java:60)
	at org.eclipse.kura.nm.configuration.NMSettingsConverter.create8021xMschapV2(NMSettingsConverter.java:208)
	at org.eclipse.kura.nm.configuration.NMSettingsConverter.build8021xSettings(NMSettingsConverter.java:140)
	at org.eclipse.kura.nm.configuration.NMSettingsConverter.buildSettings(NMSettingsConverter.java:97)
	at org.eclipse.kura.nm.NMDbusConnector.enableInterface(NMDbusConnector.java:443)
	at org.eclipse.kura.nm.NMDbusConnector.manageConfiguredInterface(NMDbusConnector.java:422)
	at org.eclipse.kura.nm.NMDbusConnector.doApply(NMDbusConnector.java:381)
	at org.eclipse.kura.nm.NMDbusConnector.lambda$0(NMDbusConnector.java:366)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at org.eclipse.kura.nm.NMDbusConnector.doApply(NMDbusConnector.java:363)
	at org.eclipse.kura.nm.NMDbusConnector.apply(NMDbusConnector.java:322)
	at org.eclipse.kura.nm.configuration.NMConfigurationServiceImpl.writeNetworkConfigurationSettings(NMConfigurationServiceImpl.java:317)
	at org.eclipse.kura.nm.configuration.NMConfigurationServiceImpl.update(NMConfigurationServiceImpl.java:218)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.felix.scr.impl.inject.methods.BaseMethod.invokeMethod(BaseMethod.java:228)
	at org.apache.felix.scr.impl.inject.methods.BaseMethod.access$500(BaseMethod.java:41)
	at org.apache.felix.scr.impl.inject.methods.BaseMethod$Resolved.invoke(BaseMethod.java:664)
	at org.apache.felix.scr.impl.inject.methods.BaseMethod.invoke(BaseMethod.java:510)
	at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:317)
	at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:307)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.invokeModifiedMethod(SingleComponentManager.java:836)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.modify(SingleComponentManager.java:791)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.reconfigure(SingleComponentManager.java:709)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.reconfigure(SingleComponentManager.java:673)
	at org.apache.felix.scr.impl.manager.ConfigurableComponentHolder.configurationUpdated(ConfigurableComponentHolder.java:435)
	at org.apache.felix.scr.impl.manager.RegionConfigurationSupport.configurationEvent(RegionConfigurationSupport.java:316)
	at org.apache.felix.scr.impl.manager.RegionConfigurationSupport$2.configurationEvent(RegionConfigurationSupport.java:118)
	at org.eclipse.equinox.internal.cm.EventDispatcher$1.run(EventDispatcher.java:92)
	at org.eclipse.equinox.internal.cm.SerializedTaskQueue$1.run(SerializedTaskQueue.java:40)
```

We should check that all the required fields are ok before submitting the form. This is usually done in the `Tab8021xUi.java::isValid` which now always returns true.

**Review by marcellorinaldo — CHANGES_REQUESTED** [review]

File: kura/org.eclipse.kura.web2/META-INF/MANIFEST.MF [hunk 6]
```diff
@@ -69,7 +69,7 @@ Import-Package: com.eclipsesource.json;version="0.9.5",
  org.eclipse.kura.net.status.ethernet;version="[1.0,2.0)",
  org.eclipse.kura.net.status.modem;version="[1.0,2.0)",
  org.eclipse.kura.net.status.wifi;version="[1.0,2.0)",
- org.eclipse.kura.net.wifi;version="[2.2,3.0)",
+ org.eclipse.kura.net.wifi;version="[2.4,2.5)",
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.web2/META-INF/MANIFEST.MF:73]: We might want to discuss this choice with the rest of team. We can keep it this way if we plan to maintain `web2` in the distribution forever.

**Comment by GregoryIvo** [comment]
@marcellorinaldo The new commits add the following validation
image
image
image
image

**Comment by GregoryIvo** [comment]
I also hid the check button for now:
image

**Review by marcellorinaldo — APPROVED** [review]

**Review by marcellorinaldo — APPROVED** [review]

**Pull request merged**